src/DynamoCore/Graph/Notes/NoteModel.cs

```diff
@@ -111,10 +122,16 @@ protected override void DeserializeCore(XmlElement nodeElement, SaveContext cont
             Text = helper.ReadString("text", "New Note");
             X = helper.ReadDouble("x", 0.0);
             Y = helper.ReadDouble("y", 0.0);
+            createdPinNode = helper.ReadBoolean("createdPinNode");
 
             // Notify listeners that the position of the note has changed, 
             // then parent group will also redraw itself.
             ReportPosition();
+            if(createdPinNode && UndoRequest != null)
```

Comment:
what's happening here is not obvious, and it seems strange to raise an event `UndoRequest` every time a note is deserialized, I know this method is currently used during undo/redo and copy paste constructors... maybe you could do a small write up in the PR description of what the issue is, and we can brainstorm other ways to fix this.

Reply:
in any case, I would definitely add a comment here if end up keeping this solution.

Reply:
I changed this code to the ActionNode method as a suggestion. Let me know your opinion about it.